### PR TITLE
Draw a grid of points, instead of lines. Increase velocity. 💅

### PR DESCRIPTION
**Pull request by emmenko**
Small refactoring to draw a better canvas (instead of using lines). 

pacman_2

I also increased the "velocity" of pacman (move by `2px` instead of `1px`) as I noticed (at least on my machine) that it was going "slower", probably due to the many points that we are rendering now.

### CODE REVIEW

File: src/index.re [hunk 1]
```diff
@@ -94,25 +99,26 @@ let getDirection = ({direction, pos: (x, y), nextDirection}) => {
...
+  let direction = getDirection({...agentState, pos: (x, y)});
   switch (direction) {
-  | Up => {direction, nextDirection, pos: (x, y -. 1.)}
-  | Down => {direction, nextDirection, pos: (x, y +. 1.)}
-  | Right => {direction, nextDirection, pos: (x +. 1., y)}
-  | Left => {direction, nextDirection, pos: (x -. 1., y)}
+  | Up => {...agentState, direction, pos: (x, y -. agentState.velocity)}
+  | Down => {...agentState, direction, pos: (x, y +. agentState.velocity)}
+  | Right => {...agentState, direction, pos: (x +. agentState.velocity, y)}
+  | Left => {...agentState, direction, pos: (x -. agentState.velocity, y)}
```
Comment (emmenko) [inline src/index.re:133]: This part is just unrelated small improvements:
* spread state
* improve variable names

File: src/index.re [hunk 2]
```diff
@@ -173,8 +179,11 @@ let getUpdatedListOfFruitsWithChance = fruits =>
     };
   };
 
-let drawFruit = (fruit: fruit, env) =>
-  Draw.(ellipsef(~center=fruit.pos, ~radx=5., ~rady=5., env));
+let drawFruit = (fruit: fruit, env) => {
+  open Draw;
+  fill(Constants.green, env);
```
Comment (emmenko) [inline src/index.re:196]: New color for the fruit

File: src/index.re [hunk 3]
```diff
@@ -193,18 +202,20 @@ let filterFruitsAndGetPoints = state =>
...
+  let (pointsX, pointsY) = GridSizes.gridPointsMatrix;
+  pointsX
+  |> List.iter(pointRow => {
+       let pointX = pointRow *. GridSizes.stepSizef;
+       pointsY
+       |> List.iter(pointColumn => {
+            let pointY = pointColumn *. GridSizes.stepSizef;
+            ellipsef(~center=(pointX, pointY), ~radx=1., ~rady=1., env);
+          });
+     });
```
Comment (emmenko) [inline src/index.re:229]: This is the main change. We iterate through the matrix and render each point as a small dot.

**Comment by ncthbrt** [comment]
Suggestion (not sure if buffers are part of the reprocessing api, but it is part of processing): Why not draw the grid to a buffer and use that instead of redrawing each time? 

Update: Looked at docs. Should be able to create an image and draw to an image. Which can be reused

https://schmavery.github.io/reprocessing/api/Reprocessing_Draw.html#value-withImage

**Comment by emmenko** [comment]
> Why not draw the grid to a buffer and use that instead of redrawing each time?

That sounds much better yeah. I'll look into the image thingy. Thanks for the useful tip! 👌

**Comment by emmenko** [comment]
@ncthbrt thanks a lot for the tip, it works like a charm! (see last commit https://github.com/reasonml-community/repacman/pull/2/commits/5b8f8edcbaf477d441b86cfb887c7c2063bc1b9d) 🙌

/cc @bsansouci pinging you since you implemented the `withImage` API, just to make sure that the usage is correct 🙏

**Comment by bsansouci** [comment]
@emmenko yup that looks right. 

So far the biggest downside to this is that it won’t hotreload any changes to the grid. That can be a bit annoying when you forget that ;)

**Comment by emmenko** [comment]
Oh I see. Well so far hot reloading kept crashing after two reloads, so I'm not using that 😅

```bash
$ yarn start
yarn run v1.5.1
warning package.json: No license field
$ ./lib/bs/bytecode/indexhot.byte
Rebuilding hotloaded module
Successfully changed functions
Rebuilding hotloaded module
Successfully changed functions
Rebuilding hotloaded module
Successfully changed functions
Rebuilding hotloaded module
File "src/index.re", line 1:
Error: I/O error: lib/bs/bytecode/src/index.re.cmi: No such file or directory
Hotreload failed
Fatal error: exception Failure("Unknown error while loading plugin")
```

**Comment by bsansouci** [comment]
@emmenko are you using vscode?
We had this issue on the example repo: https://github.com/bsansouci/reprocessing-example/pull/9

**Comment by emmenko** [comment]
Yes I am. Yep, that fixes the issue. Thanks for the link! 🙏

**Comment by emmenko** [comment]
Alright, if no one has any objections I'd like to get this merged 😇

**Comment by ncthbrt** [comment]
@emmenko Do you have commit rights?

**Review by kiraarghy — APPROVED** [review]
Seems good.

**Comment by emmenko** [comment]
🚀

**Comment by emmenko** [comment]
@ncthbrt welcome to the team! 👏

**Pull request merged**